### ISSUE TITLE
HTML support for set_test_documentation() and set_suite_documentation()

### ISSUE THREAD

**Issue by yahman72** [robotframework/robotframework#2549]
The title says it all: it would nice to have the \*HTML\* support also for the `set test documentation` and `set_suite_documentation` KW's. This would also be consistent with the `set test message` that supports the \*HTML\*.

My Use Case was:
I was trying to make links to execution logs (downloaded from SUT and stored in directories with timestamps) from the report.html. 
When I ran into problems with the `set test message` and HTML (see #2548) I tried to use `set test documentation` only to notice that it doesn't support HTML at all.

**Comment by pekkaklarck** [comment]
Notice that you can use Robot's [custom documentation formatting](http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#documentation-formatting) in docs already now. Being able to use HTML freely would be more powerful, though. A simple solution would be allowing `*HTML*` in the beginning of the documentation similarly as we allow it in exception messages. Are you interested to provide a PR?

**Comment by mcoderr** [comment]
Hello,
may i can work on this ticket

**Comment by pekkaklarck** [comment]
Go ahead!

**Comment by geekypandey** [comment]
Hi! Is this issue stale, or can I take up this issue?

**Comment by pekkaklarck** [comment]
Seems nobody is working with this so go ahead @geekypandey!

**Comment by mcoderr** [comment]
Please go head

**Comment by pekkaklarck** [comment]
This issue has some design decisions to be made that are related to #4068. The idea of that issue is to make test and suite messages available both as plain text and as HTML and there are s

We have #4068 about making test and suite messages available both as plain text and HTML. There are lot of similarities to this issue on implementation level. I just commented that issue related to [possible design](https://github.com/robotframework/robotframework/issues/4068#issuecomment-918354447).

**Comment by pekkaklarck** [comment]
In addition to `set_xxx_documentation` methods supporting HTML, we should make it possible to use HTML when defining suite, test and keyword documentation in data:

```robotframework
*** Settings ***
Documentation    *HTML* Example <b>doc</b> in HTML.
```

Most likely supporting this wouldn't make the implementation any more complicated.

**Comment by geekypandey** [comment]
okay! I will start looking in the issue.

**Comment by dadwi91** [comment]
If this is still open, I am gonna pick this up

**Comment by pekkaklarck** [comment]
Yes, this is still open. See also related #4068. Ping me on the #devel channel in our Slack if you need any help!

**Comment by pekkaklarck** [comment]
There's some recent development related to this. First of all, we are planning to add Markdown support to suite and test documentation in RF 7.5 (#5602) and adding HTML support at the same time would be great. Or do you think we still need HTML support if Markdown is supported?

Another activity is PR #5624 that implements the functionality discussed in this issue. There's a problem, though, that the design how to configure the documentation format discussed in #5602 is different to what has been discussed here. Here the idea is to [prefix docs with `*HTML*`](https://github.com/robotframework/robotframework/issues/2549#issuecomment-918362078) and in #5602 the idea is to [use file level configuration](https://github.com/robotframework/robotframework/issues/5602#issuecomment-3775464102). I think the #5602 style is better as it allows using the same format with all docs without repeating the `*HTML*` marker. Supporting such markers as a way to override the default documentation format could be considered, though.

Looking at this issue and the PR, I realized I had forgotten about `Set Suite/Test Documentation` keyword when designing #5602. The prefix approach discussed in this issue works also in that case, but adding separate `format` argument to these keywords works well too. I need to add a comment about that to #5602.

I consider Markdown support more important than HTML support and believe we should implement #5602 first. Good news is that I've already prototyped that and the plan is to start RF 7.5 development soon as well. I'll add this issue to RF 7.5 scope as well and hope that @msamo-n can wait a bit until #5602 lands.

**Comment by msamo-n** [comment]
I, personally, like #5602 👍
Supporting of Markdown would also solve my usecases, so I won't need HTML support then.

**Comment by pekkaklarck** [comment]
Yeah, it's not certain do we need raw HTML support if we support Markdown. Adding such support after #5602 is done would likely be very easy, though.

**Comment by adombeck** [comment]
> Yeah, it's not certain do we need raw HTML support if we support Markdown. Adding such support after https://github.com/robotframework/robotframework/issues/5602 is done would likely be very easy, though.

We would love to have HTML support in suite metadata (i.e. the one set with `Set Suite Metadata`). Would it be possible to also consider that keyword in #5602? I could contribute a PR after #5602 is done to add the HTML support.
